besskge/dataset.py

```diff
@@ -284,11 +284,71 @@ def build_yago310(cls, root: Path) -> "KGDataset":
             neg_tails=None,
         )
 
+    @classmethod
+    def from_triples(
+        cls,
+        data: NDArray[np.int32],
+        split: Tuple[float, float, float] = (0.7, 0.15, 0.15),
+        entity_dict: Optional[List[str]] = None,
+        relation_dict: Optional[List[str]] = None,
+        type_offsets: Optional[Dict[str, int]] = None,
+        neg_heads: Optional[Dict[str, NDArray[np.int32]]] = None,
+        neg_tails: Optional[Dict[str, NDArray[np.int32]]] = None,
+    ) -> "KGDataset":
+        """
+        Build a dataset from an array of triples.
+
+        :param data:
+            Numpy array of triples [head_id, relation_id, tail_id]. Shape
+            (num_triples, 3).
+        :param split:
+            Optional tuple to set the train/validation/test split.
+        :param entity_dict:
+            Optional entity labels by ID.
+        :param relation_dict:
+            Optional relation labels by ID.
+        :param type_offsets:
+            Offset of entity types
+        :param neg_heads:
+            Optional IDs of negative heads.
+        :param neg_tails:
+            Optional IDs of negative tails.
+
+        :return: Instance of the KGDataset class.
+        """
+        num_triples = data.shape[0]
+        num_train = int(num_triples * split[0])
+        num_valid = int(num_triples * split[1])
+
+        rng = np.random.default_rng()
```

Comment:
For reproducible results, I would suggest adding a `seed` argument to the method and passing it to the numpy rng. Also, for consistency, I have used `rng = np.random.RandomState(seed)` in other parts of the package (e.g. in besskge.sharding).

Reply:
Agree with the seed. However, `default_rng()` gives us a [random generator](https://numpy.org/doc/stable/reference/random/generator.html) which now seems to be preferred over `RandomState`. But I have no strong opinion here

Reply:
I was not aware of this, but I agree. Could I please ask you to update to `default_rng` all the occurrences of `RandomState`? Otherwise I will do it in a separate PR, no rush.